app/services/publisher_statement_getter.rb

```diff
@@ -7,6 +7,23 @@ class Statement
 
     UPHOLD_CONTRIBUTION = "uphold_contribution".freeze
     UPHOLD_CONTRIBUTION_SETTLEMENT = "uphold_contribution_settlement".freeze
+
+    def fee?
+      transaction_type == 'fees'
+    end
+
+    def eyeshade?
+      transaction_type == 'contribution_settlement' || transaction_type == 'referral_settlement' || fee?
+    end
+
+    def earning_period
+      # If the transaction_type is from Eyeshade this means the period was for the previous month
+      if eyeshade?
+        (created_at - 1.month).at_beginning_of_month
+      else
+        created_at.at_beginning_of_month
```

Comment:
We don't need the time and timezone right?

```suggestion
        created_at.to_date.at_beginning_of_month
```

Reply:
Going to leave this as-is because`to_date` was already called on `created_at` as shown in the code below

#### In the normal Eyeshade based transactions
https://github.com/brave-intl/publishers/blob/bbe717855ddb41627cdc8ded5921589b133d6d37/app/services/publisher_statement_getter.rb#L64

#### In the Uphold-based transactions
https://github.com/brave-intl/publishers/blob/bbe717855ddb41627cdc8ded5921589b133d6d37/app/services/publisher_statement_getter.rb#L83